DOC-14412 Product Change- PR #150364 - sql, settings, cli: Add sensitive, reportable columns to cluster_settings and update debug zip setting redaction logic (#20371)
cockroachdb/docs: src/current/v25.4/cockroach-debug-zip.md

`--include-running-job-traces` | Include information about each traceable job that is running or reverting (such as [backup]({% link {{ page.version.version }}/backup.md %}), [restore]({% link {{ page.version.version }}/restore.md %}), [import]({% link {{ page.version.version }}/import-into.md %}), [physical cluster replication]({% link {{ page.version.version }}/physical-cluster-replication-technical-overview.md %})) in `jobs/*/*/trace.zip` files. This involves collecting cluster-wide traces for each running job in the cluster.<br /><br />**Default:** true
`--nodes` | Specify nodes to inspect as a comma-separated list or range of node IDs. For example:<br /><br />`--nodes=1,10,13-15`

> <a id="redact"></a>`--redact` | Redact sensitive data from the generated `.zip`, with the exception of range keys, which must remain unredacted because they are essential to support CockroachDB. This flag replaces the deprecated `--redact-logs` flag, which only applied to log messages contained within `.zip`.<br><br>Cluster settings marked as "sensitive" are always redacted in the `crdb_internal.cluster_settings.txt` and `cluster_settings_history.txt` files, regardless of whether the `--redact` flag is used. When the `--redact` flag is used, the non-default values of "non-reportable" settings are also redacted.<br><br>To redact hostnames and IP addresses in `.json` files, such as `status.json`, `details.json`, and `ranges.json`, you will also need to enable the [cluster setting `debug.zip.redact_addresses.enabled`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-debug-zip-redact-addresses-enabled). Note that enabling this cluster setting will not redact all hostnames and IP addresses in the `nodes.json` and `gossip.json` files.<br><br>For examples, refer to [Redact sensitive information](#redact-sensitive-information).

Reviewer: ```suggestion
<a id="redact"></a>`--redact` | Redact sensitive data in the generated `.zip`, including non-default values of "non-reportable" settings. This flag replaces the deprecated `--redact-logs` flag, which only applied to log messages in the `.zip`. Note that some data is not redacted when `--redact` is used:<ul><li>Range keys are never redacted, since they are essential for CockroachDB support.</li><li>Cluster settings marked as "sensitive" are always redacted in `crdb_internal.cluster_settings.txt` and `cluster_settings_history.txt`.</li><li>Hostnames and IP addresses in `.json` files (such as `status.json`, `details.json`, and `ranges.json`) are only redacted if the cluster setting [`debug.zip.redact_addresses.enabled`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-debug-zip-redact-addresses-enabled) is enabled. However, some addresses in `nodes.json` and `gossip.json` are never redacted.</li></ul>For examples, refer to [Redact sensitive information](#redact-sensitive-information).
```

Author: updated

`--include-running-job-traces` | Include information about each traceable job that is running or reverting (such as [backup]({% link {{ page.version.version }}/backup.md %}), [restore]({% link {{ page.version.version }}/restore.md %}), [import]({% link {{ page.version.version }}/import-into.md %}), [physical cluster replication]({% link {{ page.version.version }}/physical-cluster-replication-technical-overview.md %})) in `jobs/*/*/trace.zip` files. This involves collecting cluster-wide traces for each running job in the cluster.<br /><br />**Default:** true
`--nodes` | Specify nodes to inspect as a comma-separated list or range of node IDs. For example:<br /><br />`--nodes=1,10,13-15`

> <a id="redact"></a>`--redact` | Redact sensitive data in the generated `.zip` file. This flag replaces the deprecated [`--redact-logs`](#redact-logs) flag.<br><br>This flag redacts the following data:<ul><li>Sensitive data in log messages. Refer to [Log redaction](#log-redaction) for an example.</li><li>Non-default values of cluster settings marked as **not** `reportable` in `crdb_internal.cluster_settings.txt` and `cluster_settings_history.txt`. Refer to [Cluster settings redactions](#cluster-settings-redaction) for an example.</li><li>Hostnames and IP addresses in `.json` files (such as `status.json`, `details.json`, and `ranges.json`) when the cluster setting [`debug.zip.redact_addresses.enabled`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-debug-zip-redact-addresses-enabled) is enabled. Refer to [Hostname and IP address redaction](#hostname-and-ip-address-redaction) for an example.</li></ul>This flag does **not** affect the following data:<ul><li>Range keys are **never** redacted because they are essential for CockroachDB support.</li><li>Cluster settings marked as `sensitive` are **always** redacted in `crdb_internal.cluster_settings.txt` and `cluster_settings_history.txt`. For an example, refer to [Cluster settings redactions](#cluster-settings-redaction).</li><li>Some hostnames and IP addresses in the `nodes.json` and `gossip.json` files are **never** redacted, even when `debug.zip.redact_addresses.enabled` is enabled.</li></ul>For examples, refer to [Redact sensitive information](#redact-sensitive-information).

Author: good catch! updated

server/server.go:1423 ⋮ password of user ‹×› was set to ‹×›
~~~

#### Cluster settings redaction

Example of a cluster setting in `crdb_internal.cluster_settings.txt` without redaction enabled:

~~~
variable	value	type	public	sensitive	reportable	description	default_value	origin
...

> cluster.organization	Cockroach Labs Testing	s	t	f	f	organization name		override

Reviewer 2: nit: its kinda hard to read this table in the rendered markdown. Is it possible to make the columns and headers align better?

server/server.go:1423 ⋮ password of user ‹×› was set to ‹×›
~~~

#### Cluster settings redaction

Example of a non-reportable cluster setting in `crdb_internal.cluster_settings.txt` without redaction enabled:

~~~
variable             value                  type public sensitive reportable description       default_value origin
...
cluster.organization Cockroach Labs Testing s    t      f         f          organization name               override
~~~

Enable log redaction:

{% include_cached copy-clipboard.html %}
~~~ shell
$ cockroach debug zip ./cockroach-data/logs/debug.zip --redact --insecure --host=200.100.50.25
~~~

> A non-reportable cluster setting in `crdb_internal.cluster_settings.txt` with redaction enabled:

Reviewer: ```suggestion
The same non-reportable cluster setting in `crdb_internal.cluster_settings.txt` with its value redacted:
```

Author: updated

server/server.go:1423 ⋮ password of user ‹×› was set to ‹×›
~~~

#### Cluster settings redaction

> Example cluster settings in `crdb_internal.cluster_settings.txt` without redaction enabled:

Author: updated

~~~
variable                          value                  type public sensitive reportable description                                   default_value origin
...
cluster.organization              Cockroach Labs Testing s    t      f         f          organization name                                           override
...
server.identity_map.configuration <redacted>             s    t      t         f          system-identity to database-username mappings               default
~~~

`server.identity_map.configuration` is redacted since `sensitive` equals `true`.

Enable log redaction:

{% include_cached copy-clipboard.html %}
~~~ shell
$ cockroach debug zip ./cockroach-data/logs/debug.zip --redact --insecure --host=200.100.50.25
~~~

> Cluster settings in `crdb_internal.cluster_settings.txt` with redaction enabled:

Author: updated

~~~
variable                          value      type public sensitive reportable description                                    default_value origin
...
cluster.organization              <redacted> s    t      f         f          organization name                                            override
...
server.identity_map.configuration <redacted> s    t      t         f          system-identity to database-username mappings                default
~~~

> `server.identity_map.configuration` is still redacted. `cluster.organization` is now redacted since `reportable` equals `false` and the value of `Cockroach Labs Testing` does not equal the default (in this case, the empty string).

Author: updated

#### Hostname and IP address redaction

> Example of `status.json` without hostname and IP address redaction enabled:

Author: updated
